### PR TITLE
[Google Blockly] Disable modal function editor for Dance and Poetry levels

### PR DESCRIPTION
**Pull request by mikeharv**
The modal function editor is a custom feature for Blockly labs that lives on our old Blockly fork. It does not currently work with Google Blockly labs (e.g. Poetry). As of today, no Poetry levels use functions, but the levels have all been created with a default setting to set `use_modal_function_editor` to true. Because this state doesn't match the actual experience, we'd like to set this property to false on all Poetry levels. All changes to level files here are just swapping that boolean value.

Additionally, we'd like to update this default value for Poetry levels and prevent levelbuilders from setting the value to true (at least until we have a working modal editor). The change to the .haml file removes the highlighted option for the lab types we care about. 
image

The next big Blockly milestone is to move Dance over to Google Blockly. We will be also doing this before adding support for the modal editor so we'd like to make the same changes here. The exception is that we are not going to update any Dance levels. This is because some Dance levels do use the (cdo Blockly) modal function editor. The team is comfortable with disabling this editor away in Dance levels when we switch to Google Blockly but we don't want to have two separate UI changes. The current plan is to make a similar set of updates to Dance levels when we switch Blockly versions. This is drafted here: https://github.com/code-dot-org/code-dot-org/pull/48860/

I'm requesting review from @code-dot-org/studentlearning as usual and adding @code-dot-org/teacher-tools just to weigh in on whether there are any risks to updating the levels or models like this.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/views/levels/editors/fields/_blockly.html.haml [hunk 1]
```diff
@@ -18,8 +18,9 @@
     = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :is_k1, description: "Is Pre-Reader level"}
     = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :disable_variable_editing, description: "Disable editing of variable names and creation of new variables"}
     = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :show_type_hints, description: "Show a small visual hint for what type of block a connection accepts"}
-
-  = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :use_modal_function_editor, description: "Use modal function editor"}
+    -# Google Blockly does not currently support the modal function editor
+  - unless (@level.is_a?(Dancelab) || @level.is_a?(Poetry))
```
Comment (bencodeorg) [inline dashboard/app/views/levels/editors/fields/_blockly.html.haml:22]: Do we have any Rails-side knowledge of whether a lab uses Google or CDO blockly by default?

**Review by bencodeorg — APPROVED** [review]
LGTM! How did you come up with the list of poetry levels to edit? Were these manual changes to the .level files, or did you do it in some automated fashion?

File: dashboard/app/views/levels/editors/fields/_blockly.html.haml (hunk 1 above)
Comment (maddiedierker) [reply dashboard/app/views/levels/editors/fields/_blockly.html.haml:22]: we do! every Level can respond to `uses_google_blockly?`. the default is defined in level.rb:

https://github.com/code-dot-org/code-dot-org/blob/aabc451911920d8e60d98963995f8416c5cd29f2/dashboard/app/models/levels/level.rb#L608-L610

and individual level types can override it:

https://github.com/code-dot-org/code-dot-org/blob/667d7c8941f9fd1786f2f6c75cf0f43326c61f3a/dashboard/app/models/levels/poetry.rb#L76-L78

Comment (bencodeorg) [reply dashboard/app/views/levels/editors/fields/_blockly.html.haml:22]: Nice, good to know! I guess in this case we're not officially on google blockly in dance, so I can see why it's written this way for now. Maybe we can note somewhere that we should update this (maybe already noted in Mike's head :)) when we flip to google blockly in dance to just use that method?

Comment (mikeharv) [reply dashboard/app/views/levels/editors/fields/_blockly.html.haml:22]: Yup, I linked this draft PR in the description above that includes making this change to the dance.rb file: https://github.com/code-dot-org/code-dot-org/pull/48860

Comment (bethanyaconnor) [inline dashboard/app/views/levels/editors/fields/_blockly.html.haml:22]: Not something that needs to be handled in this PR but would it makes sense to have a method to check this on each level? Something like `@level.supports_google_blockly?`?

**Review by bethanyaconnor — APPROVED** [review]

File: dashboard/app/views/levels/editors/fields/_blockly.html.haml (hunk 1 above)
Comment (mikeharv) [reply dashboard/app/views/levels/editors/fields/_blockly.html.haml:22]: Yes, I think this is possible: https://github.com/code-dot-org/code-dot-org/pull/48857#discussion_r1006253039

Right now that logic wouldn't quite get us where we need as Dance isn't on Google Blockly _yet_. The current plan is to switch to that method once we flip Dance to Google Blockly. This is being tracked in the draft PR that was started for the actual flip.

**Pull request merged**